sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -118,12 +117,8 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
       val rdd = v1Relation.buildScan()
       val unsafeRowRDD = DataSourceStrategy.toCatalystRDD(v1Relation, output, rdd)
 
-      val tableIdentifier = v2Relation.identifier.map(_.asMultipartIdentifier) match {
-        case Some(Seq(schema, tableName)) =>
-          Some(new TableIdentifier(tableName, Some(schema), v2Relation.catalog.map(_.name())))
-        case _ =>
-          None
-      }
+      val catalogName = v2Relation.catalog.map(_.name())
+      val tableIdentifier = v2Relation.identifier.flatMap(_.asTableIdentifier(catalogName))
```

Comment:
I think we can simplify this to `v2Relation.identifier.flatMap(_.asTableIdentifier.copy(catalog = catalogName)))`

Reply:
then we don't need to add a new method.

Reply:
I don't want to use asTableIdentifier, it can throw an exception

Reply:
If there are multiple namespaces

Reply:
isn't it a better behavior? Silently ignore unexpected namespace is risky.

Reply:
This is only for telemetry and Spark UI, before initial change (one month ago), table identifier was null. IMO we should not risk regression for the purpose of telemetry and observability, whenever we get multipart identifier, the exception will be thrown